docs/spec/typeddict.rst

```diff
@@ -128,95 +244,232 @@ key, and the ``'name'`` key is missing::
 
-    class Y(TypedDict):
-        y: str
+If an item is mutable in a superclass, it must remain mutable in the subclass. Similarly,
+mutable items that are :term:`required` in a superclass must remain required in the subclass,
+and mutable :term:`non-required` items in the superclass must remain non-required in the subclass.
+However, if the superclass item is :term:`read-only`, a superclass item that is non-required
+may be overridden with a required item in the subclass. A read-only item in a superclass
+may be redeclared as mutable (that is, without the ``ReadOnly`` qualifier) in a subclass.
+These rules are necessary for type safety.
 
-    class XYZ(X, Y):
-        z: bool
+If an item is read-only in the superclass, the subclass may redeclare it with a different type
+that is :term:`assignable` to the superclass type. Otherwise, changing the type of an item is not allowed.
+Example::
 
-The TypedDict ``XYZ`` has three items: ``x`` (type ``int``), ``y``
-(type ``str``), and ``z`` (type ``bool``).
+   class X(TypedDict):
+       x: str
+       y: ReadOnly[int]
 
-A TypedDict cannot inherit from both a TypedDict type and a
-non-TypedDict base class other than ``Generic``.
+   class Y(X):
+       x: int  # Type check error: cannot overwrite TypedDict field "x"
+       y: bool  # OK: bool is assignable to int, and a mutable item can override a read-only one
 
-Additional notes on TypedDict class inheritance:
+Openness
+^^^^^^^^
 
-* Changing a field type of a parent TypedDict class in a subclass is not allowed.
-  Example::
+The openness of a TypedDict (whether it is :term:`open`, :term:`closed`, or has :term:`extra items`)
+is inherited from its superclass by default::
 
-   class X(TypedDict):
-      x: str
+    class ClosedBase(TypedDict, closed=True):
+        name: str
 
-   class Y(X):
-      x: int  # Type check error: cannot overwrite TypedDict field "x"
+    class ClosedChild(ClosedBase):  # also closed
+        pass
+
+    class ExtraItemsBase(TypedDict, extra_items=int | None):
+        name: str
+
+    class ExtraItemsChild(ExtraItemsBase):  # also has extra_items=int | None
+        pass
+
+However, subclasses may also explicitly use the ``closed`` and ``extra_items`` arguments
+to change the openness of the TypedDict, but in some cases this yields a type checker error.
+If the base class is open, all possible states are allowed in the subclass: it may remain open,
+it may be closed (with ``closed=True``), or it may have extra items (with ``extra_items=...``).
+If the base class is closed, any child classes must also be closed.
+If the base class has extra items, but they are not read-only, the child class must also allow
+the same extra items. If the base class has read-only extra items, the child class may be closed,
+or it may redeclare its extra items with a type that is :term:`assignable` to the base class type.
+Child classes may also have mutable extra items if the base class has read-only extra items.
```

Comment:
Organizing this section into bullets might help with readability, since it exhausts all possible openness a base TypedDict class can have.